### PR TITLE
Add option to send denial email citing embargo policy

### PR DESCRIPTION
**Pull request by benzuzu**
### Embargo Country Text Match will Trigger Warning in Slack Integration

If a user registers through an organization located in an embargoed country (list specified in application properties), a warning will appear in the Slack request:

Screen Shot 2021-07-14 at 11 49 05 AM

### Embargo Email

There is now an option in Slack to deny a user due to embargo policy. The following email will be sent to the user:

Screen Shot 2021-07-20 at 12 12 24 PM

### Changes to Slack Integration

The Slack integration has been adjusted accordingly.

- Non-approve buttons have been moved to a dropdown menu for organization
- Clarification and Denial emails can be sent in this menu

Screen Shot 2021-07-14 at 11 41 37 AM
Screen Shot 2021-07-14 at 11 41 48 AM

### CODE REVIEW

File: src/main/java/org/mskcc/cbio/oncokb/domain/UserStatusChecks.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ */
+public class UserStatusChecks {
+    UserDTO userDTO;
+
+    boolean trialAccountActivated;
+    boolean trialAccountInitiated;
+    boolean academicClarificationEmailSent;
+    boolean embargoEmailSent;
+
+    public UserStatusChecks(UserDTO userDTO, UserService userService, SlackService slackService) {
```
Comment (zhx828) [inline src/main/java/org/mskcc/cbio/oncokb/domain/UserStatusChecks.java:18]: We should not pass the service layout objects directly but the actual value to initialize the UserStatusChecks

File: src/main/java/org/mskcc/cbio/oncokb/config/application/ApplicationProperties.java [hunk 2]
```diff
@@ -171,6 +176,10 @@ public void setFrontend(FrontendProperties frontend) {
         return getDomains(this.getTrialedDomains());
     }
 
+    public List<String> getEmbargoedCountriesList() {
+        return getDomains(this.getEmbargoedCountries());
+    }
+
     private List<String> getDomains(String domainStr) {
```
Comment (zhx828) [inline src/main/java/org/mskcc/cbio/oncokb/config/application/ApplicationProperties.java:183]: Maybe we should rename this method now  to `stringSplitAndTrim`? And good name recommendation?

**Review by zhx828 — CHANGES_REQUESTED** [review]
Looks good, just some minor tweaks. Let's also discuss the issue on next Tuesday meeting.

File: src/main/java/org/mskcc/cbio/oncokb/config/application/ApplicationProperties.java (hunk 2 above)
Comment (benzuzu) [reply src/main/java/org/mskcc/cbio/oncokb/config/application/ApplicationProperties.java:183]: How does `toList` sound? Or `stringToList`, `getList`

**Comment by benzuzu** [comment]
Closed until further notice

**Pull request closed without merging**